tensorcircuit/translation.py

```diff
@@ -219,6 +219,47 @@ def qir2qiskit(
     return qiskit_circ
 
 
+def _translate_qiskit_params(gate_info, binding_params):
+    parameters = []
+    for p in gate_info[0].params:
+        from qiskit.circuit.parametervector import ParameterVectorElement
+        from qiskit.circuit import Parameter, ParameterExpression
+
+        if isinstance(p, ParameterVectorElement):
+            parameters.append(binding_params[p.index])
+        elif isinstance(p, Parameter):
+            parameters.append(binding_params[p])
+        elif isinstance(p, ParameterExpression):
+            if len(p.parameters) != 1:
+                raise ValueError(
+                    f"Can't translate parameter expression with more than 1 parameters: {p}"
+                )
+            p_real = list(p.parameters)[0]
+            if not isinstance(p_real, ParameterVectorElement):
+                raise TypeError(
+                    "Parameters in parameter expression should be ParameterVectorElement"
+                )
+            # note "sym" != "sim"
+            expr = p.sympify().simplify()
+            # only allow simple expressions like 1.0 * theta
+            if not expr.is_Mul:
```

Comment:
is there any common case for builtin qiskit ansatz beyond the simple 1*param style that you ever came across?

Reply:
In UCCSD, the same parameter theta is used in multiple gates and there are both -1 * theta and 1 * theta. Converting all of them to 1 * theta will lose the phase information.

Reply:
good point to against parameterexpression -> jax tensor idea! the connection between parameters will lose